File: kinotic-js/workspace/packages/os-api/src/api/model/KinoticClusterInfo.ts
import { KinoticNodeInfo } from '@/api/model/KinoticNodeInfo'

/**
 * The kinotic-server cluster's current topology and state. Mirrors the server's
 * {@code org.kinotic.domain.api.model.cluster.KinoticClusterInfo}.
 */
export class KinoticClusterInfo {
    public localNodeId: string = ''
    public serverNodeCount: number = 0
    public topologyVersion: number = 0
    public clusterState: string = ''
    public nodes: KinoticNodeInfo[] = []
    public active: boolean = false
}

File: kinotic-system-api/src/main/java/org/kinotic/system/api/model/cluster/KinoticClusterInfo.java
package org.kinotic.system.api.model.cluster;

import lombok.Builder;
import lombok.Data;

import java.util.List;

/**
 * Contains information about the Ignite cluster.
 */
@Data
@Builder
public class KinoticClusterInfo {

    /**
     * The unique identifier of the local node.
     */
    private String localNodeId;
    
    /**
     * The total number of server nodes in the cluster.
     */
    private int serverNodeCount;
    
    /**
     * The current topology version of the cluster.
     */
    private long topologyVersion;
    
    /**
     * The current state of the cluster (e.g., ACTIVE, INACTIVE).
     */
    private String clusterState;
    
    /**
     * Information about all server nodes in the cluster.
     */
    private List<KinoticNodeInfo> nodes;
    
    /**
     * Indicates whether the cluster is active.
     */
    private boolean active;
}
